CIP-????/README.md

```diff
@@ -0,0 +1,250 @@
+
+- _Pineapples on pizza?_
+  - [ ] yes
+  - [ ] no
+
+The serialised question and answers will be posted on-chain and signed by one of the delegate genesis keys owned by The Cardano Foundation. Answers will be provided on-chain by participating SPOs via transaction metadata referring to:
+
+- The question and answers
+- The index of the chosen answer from the available choices
+- A signature from the SPO's current VRF credentials or Ed25519 declared cold credentials.
+
+> **Note**
+> In this document, every time we refer to a _serialized object_, we refer to its **canonical** CBOR representation. In particular, keys in a map are always ordered alphabetically.
+
+### Question structure
+
+A question is posted in a transaction's metadata using the metadata label `94` and the following metadata structure:
+
+```cbor
+question =
+  { 0: prompt
+  , 1: [ * choice ]
+  , ? "_": nonce
+  }
+
+prompt =
+  [ * text .size (0..64) ]
+
+choice =
+  [ * text .size (0..64) ]
+
+nonce =
+  uint
+```
+
+A nonce is optionally included to provide non-replayability should the same question and answers be asked multiple times over different periods. The transaction carrying a question **must** be signed by one of the genesis delegate keys to be considered valid. This genesis key signature isn't captured in the metadata but in the transaction itself as an extra signatory.
+
+For example, in [CBOR-diagnostic](https://www.rfc-editor.org/rfc/rfc8949#name-diagnostic-notation) notation:
+
+```cbor
+{ 94:
+  { 0: "Pineapples on pizza?"
+  , 1:
+    [ "yes"
+    , "no"
+    ]
+  }
+}
+```
+
+Which serialises down to:
+
+```
+A1185EA2007450696E656170706C6573206F6E2070697A7A613F018263796573
+626E6F
+```
+
+### Answer structure
+
+Similarly, an answer to a question is posted as transaction's metadata using the label `94` and the following metadata structure:
+
+```cbor
+answer =
+  { 2: question_hash
+  , 3: choice
+  , ? 4: vrf_witness
+  , ? 5: ed25519_witness
+  }
+
+question_hash =
+  bytes .size 32
+
+vrf_witness =
+  [ $vrf_vkey, vrf_proof ]
+
+vrf_proof =
+  [ bytes .size 64, bytes .size 16 ]
+
+ed25519_witness =
+  [ $vkey, $signature ]
+```
+
+Some remarks:
+
+1. The field `2` (`question_hash`) is a blake2b-256 hash digest, whose preimage is the entire serialised question metadata payload (with the `94` top-level label).
+1. The fields `4` and `5` are not mutually optional; said differently, at least one of the two must be provided.
+1. The `$vrf_vkey`, `$vkey` and `$signature` are as defined in the original [Cardano ledger specifications](https://github.com/input-output-hk/cardano-ledger/blob/master/eras/babbage/test-suite/cddl-files/real/crypto.cddl) with the usual rules applying to transaction metadata (i.e. byte string longer than 64 bytes are encoded as lists)
+1. In either case (VRF or Ed25519), the witness' signature is calculated from the blake2b-256 hash digest of the serialised `answer` restricted to fields `2` and `3`.
```

Comment:
Can we please be _much_ more explicit about what we do with the VRF.

A VRF is not a digital signature scheme, it's a verifiable random function. So please don't refer to VRF signatures. There's no such thing. There is a VRF _proof_. And the VRF needs an input. Lets be explicit about what the input is.

And we should be clear that we are abusing/reusing the VRF for the purpose of proving knowledge of the VRF secret, and that we are ignoring the VRF output: we only care that the proof is valid.

I would separate out the specification of how to construct and verify the witness for the Ed25519 case and the VRF case. In particular it's really important to not confuse people into thinking we use the VRF key as if it were an ED25519 key (['coz that's really bad!](https://github.com/input-output-hk/blogpost_breaking_ed25519_vrf/blob/main/blog.md)).

Reply:
Right, there are a few language abuse in the CIP regarding signature vs proofs; I've been much more careful about these formulation in the cardano-cli changes but it's a bit clumsy in the CIP. I'll edit that. 

> @dcoutts: And we should be clear that we are abusing/reusing the VRF for the purpose of proving knowledge of the VRF secret

I thought this was already clear as we're pointing back to CIP-0022's discussions where it's pretty much stated as-is; but point taken, I'll make that more apparent. 

Note that I haven't a particular attachment or preference to supporting VRF proofs here, but it's something that was explicitly raised by a few SPOs as more practical for them than accessing their cold key. I haven't pushed back on that based on Iñigo's comments on CIP-0022 which didn't really advocate _against_ the (ab)use of VRF for proving knowledge of the secret key, but rather pointed at its inefficiency. Since efficiency isn't a particular concern for this use case here either it seemed like _an acceptable option_.

Reply:
Upon further review and taking into consideration the blog post shared by @iquerejeta, I concur with @dcoutts' perspective.

The primary issue with this CIP is that it combines two cryptographic systems insecurely, even though it might be beneficial to have the option to utilize both. As suggested in the blog post, this could be resolved by segregating the domains and incorporating padding into the hashing process to mitigate potential attacks.

My primary concern is that an implementor other than IO might refer to this CIP and inadvertently apply it incorrectly, thereby exposing private keys. This risk can be averted by either opting for a VRF proof or an Ed25519 signature, or by employing an alternative setup, such as registering another Ed25519 key as outlined below. As it currently stands, this standard introduces the possibility of insecure handling of sensitive information.